packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -89,6 +89,11 @@ class MediaQueryData {
   /// system UI (such as the system notification area), or or physical
   /// intrusions in the display (e.g. overscan regions on television screens or
   /// phone sensor housings).
+  ///
+  /// If you consumed this padding (e.g. by building a widget that envelops or
+  /// accounts for this padding in its layout), you should remove this padding
+  /// for subsequent descendents in the widget tree by inserting a new
+  /// [MediaQuery] widget using the [MediaQuery.removePadding] factory.
```

Comment:
We should suggest `SafeArea` here for the straightforward case where you just need to apply padding and remove it.

Reply:
Ah man, TIL. I totally replicated that work in some places.